# Boileau_2023_GenomeBiology
Code associated with publication

>Differentialgeneanalysis_DESeq.R

This script was used to perform pairwise differential gene expression analysis between RNAseq examples.

>fig1andsupps.R

Generation of Fig1 supplemental plots including qPCR, histone western, and growth assays. Includes code to perform their statistic analysis. 

>RNASeqanalysisv2_revisions.R

Code used to further analyze and visualize RNAseq data from differentialgeneanalysis.

>chromhmm_permutationtestfinal.R

Function and code utilized to perform statistical tests on nearby transcription for chromhmm states in Fig.5B and 5D.

>epigenome_master_k4m1_v2.R

Processing and analysis scripts for h3k4me1 data and analysis using h3k4me1 peaks. Primarily Fig.2.

>combipeak_analysis.R

Similar code to epigenome_master_k4m1 but focused on H3K27ac peaks. Underlies primarily Fig.3.

>Clusterprofiler_nearTSS.R

Code used to perform gene ontology analysis on differential gene lists using clusterProfiler.

>nearestTSS_final.R

Code used to generate all nearest neighbor analysis data. Draws from dataframes created with rnaseq and k4m1/k27ac related scripts. Also used to generate enhancer compensation figures in Fig.5.

>fig4filtering.R

Generates pie charts shown in Fig.4A.

>revision_nearestTSS_dorighi2017.R

Script used for nearest neighbor analysis of MLL3/4 chip-seq peaks from Dorighi et al 2017.

>Chromhmm_analysisv2.R

chromhmm state processing and plotting for fig.5A and 5C

![image](https://user-images.githubusercontent.com/40179753/214230091-655de629-844b-4c08-af6f-03941309bb4d.png)
